drivers/bbram/bbram_stm32.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2022 Google Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_stm32_bbram
+
+#include <errno.h>
+
+#include <zephyr/drivers/bbram.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/util.h>
+LOG_MODULE_REGISTER(bbram, CONFIG_BBRAM_LOG_LEVEL);
+
+#define STM32_BKP_REG_INDEX(offset)	 ((offset) >> 2)
+#define STM32_BKP_REG_BYTE_INDEX(offset) ((offset)&0x3UL)
+#define STM32_BKP_REG(i)		 (((volatile uint32_t *)config->base_addr)[(i)])
+
+/** Device config */
+struct bbram_stm32_config {
+	/** BBRAM base address */
+	uintptr_t base_addr;
+	/** BBRAM size (Unit:bytes) */
+	int size;
+};
+
+static int bbram_stm32_read(const struct device *dev, size_t offset, size_t size, uint8_t *data)
+{
+	const struct bbram_stm32_config *config = dev->config;
+	uint32_t reg, begin, to_copy;
+
+	if (size < 1 || offset + size > config->size) {
+		return -EFAULT;
+	}
+
+	for (size_t read = 0; read < size; read += to_copy) {
```

Comment:
`to_copy` value is indeterminate.

Reply:
In line 40 we assign number of bytes we want to copy from register, so `read += to_copy` in line 37 is valid.

Reply:
@semihalf-duda-patryk , what about the first iteration? It is not guaranteed that `to_copy` is `0`. Therefore, `read += to_copy` might end up with `read` being invalid.

Reply:
The first iteration is also correct. The third statement in for loop (`read += to_copy`) is executed at the end of iteration (after the loop body).

Reply:
indeed, my bad; sorry